src/core/gltf2importer/textureparser.cpp

```diff
@@ -68,34 +112,59 @@ bool TextureParser::parse(const QJsonArray &texturesArray, GLTF2ContextPrivate *
                 sourceValue = ddsExtension[KEY_SOURCE];
             }
         }
+
         if (sourceValue.isUndefined()) {
             qCWarning(kuesa, "Unknown image source for texture");
         } else {
             const auto image = context->image(sourceValue.toInt());
-            if (image.url.isEmpty())
+            if (image.url.isEmpty() && image.data.isEmpty())
                 return false; // Not a valid image
 
             auto texture2d = std::unique_ptr<Qt3DRender::QAbstractTexture>(nullptr);
-            if (isDDSTexture) {
-                auto textureLoader = new Qt3DRender::QTextureLoader;
-                texture2d.reset(textureLoader);
-                textureLoader->setSource(image.url);
+            if (image.data.isEmpty()) {
+                if (isDDSTexture) {
+                    auto textureLoader = new Qt3DRender::QTextureLoader;
+                    texture2d.reset(textureLoader);
+                    textureLoader->setSource(image.url);
+                } else {
+                    texture2d.reset(new Qt3DRender::QTexture2D);
+
+                    auto *textureImage = sharedImages.value(image.url);
+                    if (textureImage == nullptr) {
+                        auto ti = new Qt3DRender::QTextureImage();
```

Comment:
Would checking here if whether we should be using data or url actually reduce code duplication?